gamestonk_terminal/stocks/due_diligence/dd_controller.py

```diff
@@ -148,52 +158,251 @@ def call_quit(self, _):
+
+    def call_rot(self, other_args: List[str]):
+        """Process rot command"""
+        parser = argparse.ArgumentParser(
+            add_help=False,
+            prog="rot",
+            description="""
+                Rating over time. [Source: Finnhub]
+            """,
+        )
+        try:
+            ns_parser = parse_known_args_and_warn(parser, other_args)
+            if not ns_parser:
+                return
+
+            finnhub_view.rating_over_time(ticker=self.ticker)
+
+        except Exception as e:
+            print(e, "\n")
 
     def call_rating(self, other_args: List[str]):
         """Process rating command"""
-        financial_modeling_prep_view.rating(other_args, self.ticker)
+        parser = argparse.ArgumentParser(
+            add_help=False,
+            prog="rating",
+            description="""
+                Based on specific ratios, prints information whether the company
+                is a (strong) buy, neutral or a (strong) sell. The following fields are expected:
+                P/B, ROA, DCF, P/E, ROE, and D/E. [Source: Financial Modeling Prep]
+            """,
+        )
+        parser.add_argument(
+            "-n",
+            "--num",
+            action="store",
+            dest="n_num",
+            type=check_positive,
+            default=10,
+            help="number of regions to plot that show highest interest.",
+        )
+
+        try:
+            if other_args:
+                if "-" not in other_args[0]:
+                    other_args.insert(0, "-n")
+
+            ns_parser = parse_known_args_and_warn(parser, other_args)
+            if not ns_parser:
+                return
+
+            fmp_view.rating(
+                ticker=self.ticker,
+                num=ns_parser.n_num,
+            )
+
+        except Exception as e:
+            print(e, "\n")
 
     def call_sec(self, other_args: List[str]):
         """Process sec command"""
-        market_watch_view.sec_fillings(other_args, self.ticker)
+        parser = argparse.ArgumentParser(
+            add_help=False,
+            prog="sec",
+            description="""
+                Prints SEC filings of the company. The following fields are expected: Filing Date,
+                Document Date, Type, Category, Amended, and Link. [Source: Market Watch]
+            """,
+        )
+        parser.add_argument(
+            "-n",
+            "--num",
+            action="store",
+            dest="n_num",
+            type=check_positive,
+            default=5,
+            help="number of latest SEC filings.",
+        )
+
+        try:
+            if other_args:
+                if "-" not in other_args[0]:
+                    other_args.insert(0, "-n")
+
+            ns_parser = parse_known_args_and_warn(parser, other_args)
+            if not ns_parser:
+                return
+
+            marketwatch_view.sec_filings(
+                ticker=self.ticker,
+                num=ns_parser.n_num,
```

Comment:
in the view - could we clip the link to just be the docid ?

Reply:
I prefer the whole link because it goes direct to the document